servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -1870,7 +1880,19 @@ void executeDeleteMembership(ResourceContext ctx, String domainName, String role
 
                 // process our delete role member operation
 
-                if (!con.deleteRoleMember(domainName, roleName, normalizedMember, principal, auditRef)) {
+                Role role = con.getRole(domainName, roleName);
+                boolean pending = (role != null) && (role.getReviewEnabled() == Boolean.TRUE || role.getAuditEnabled() == Boolean.TRUE);
```

Comment:
I've been thinking about this and we should introduce a new setting otherwise some may not like a change in behavior. There is also another feature request to protect deletion of roles themselves so we might be able to use the same setting for that as well.

Let's introduce a new meta attribute called deleteProtection. so for audit and review enabled roles, we'll ask for delete confirmation only if this flag is also turned on.